ignite/contrib/handlers/base_logger.py

```diff
@@ -136,14 +173,20 @@ def key_str_tf(tag: str, name: str, *args: str) -> str:
         return metrics_state_attrs_dict
 
 
-class BaseWeightsScalarHandler(BaseHandler):
+class BaseWeightsScalarHandler(BaseWeightsHandler):
     """
-    Helper handler to log model's weights as scalars.
+    Helper handler to log model's weights or gradients as scalars.
     """
 
-    def __init__(self, model: nn.Module, reduction: Callable = torch.norm, tag: Optional[str] = None):
-        if not isinstance(model, torch.nn.Module):
-            raise TypeError(f"Argument model should be of type torch.nn.Module, but given {type(model)}")
+    def __init__(
+        self,
+        model: nn.Module,
+        reduction: Callable[[torch.Tensor], torch.Tensor] = torch.norm,
```

Comment:
`Callable[[torch.Tensor], torch.Tensor]` is incorrect type hint as it can be 0D tensor or a number, see line 199 (below):
```python
if not (isinstance(o, numbers.Number) or _is_0D_tensor(o)):
```